test_nettoyage.py

```diff
@@ -0,0 +1,296 @@
+                                 ["15", "Objet3", "K", "L", "M", "", "", 0, "3", "Date!!="]
+                                 ], columns=["id", "objet", "col1", "col2", "col3", "col1Modification", "col3Modification", "booleanModification", "id_technique", "datePublicationDonnees"])
+        dict_modification = {"col1Modification": "col1",
+                             "col3Modification": "col3"
+                             }
+        _ = nettoyage.regroupement_marche(df_input, dict_modification)
+        df_attendu = pd.DataFrame([["13456", "Objet1", "D", "B", "E", "D", "E", 1.0, "0", "Date=", "2", "2"],
+                                   ["245", "Objet2", "I", "G", "J", "I", "J", 1.0, "1", "Date!=", "1", "1"],
+                                   ["134567", "Objet1", "N", "L", "M", "N", "", 1.0, "2", "Date=", "2", "2"],
+                                   ["15", "Objet3", "K", "L", "M", "", "", 0.0, "3", "Date!!=", "", "3"]
+                                   ], columns=["id_source", "objet", "col1", "col2", "col3", "col1Modification", "col3Modification", "booleanModification", "id_technique", "datePublicationDonnees", "idtech", "id"])
+        assert_frame_equal(df_attendu, df_input)
+
+    def test_regroupement_marche_complet(self):
+        df_input = pd.DataFrame([["Objet1", "2", "Date=", 2000, 1],
+                                 ["Objet2", "1", "Date!=", 2000, 2],
+                                 ["Objet1", "4", "Date=", 2000, 3],
+                                 ["Objet3", "3", "Date!!=", 3000, 4],
+                                 ["Objet3", "5", "Date!!=", 30001, 5]
+                                 ], columns=["objet", "id", "datePublicationDonnees", "montant", "useless"])
+
+        df_output = nettoyage.regroupement_marche_complet(df_input)
+        df_attendu = pd.DataFrame([["Objet1", "4", "Date=", 2000, 1, 2],
+                                   ["Objet2", "1", "Date!=", 2000, 2, 1],
+                                   ["Objet1", "4", "Date=", 2000, 3, 2],
+                                   ["Objet3", "3", "Date!!=", 3000, 4, 1],
+                                   ["Objet3", "5", "Date!!=", 30001, 5, 1]
+                                   ], columns=["objet", "id", "datePublicationDonnees", "montant", "useless", "nombreTitulaireSurMarchePresume"])
+        assert_frame_equal(df_output, df_attendu)
+
+    def test_manage_titulaires(self):
+        # Récupérationd des données réelles
+        # Les 2 fonctions ci dessous sont testées
+        df = nettoyage.manage_modifications(data)
+        df = nettoyage.regroupement_marche_complet(df)
+        # Début du test
+        df_input = df.reset_index()
+        taille_input = df_input.shape
+        nb_ligne_nulle = sum(df_input.titulaires.isna() & df_input.concessionnaires.isna())
+        df_output = nettoyage.manage_titulaires(df_input)
+        nb_ligne_finale = sum(df_input.titulaires.apply(len)) - nb_ligne_nulle
```

Comment:
```suggestion
        nb_ligne_finale = sum(df_input.titulaires.str.len()) - nb_ligne_nulle
```
cf [doc pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.len.html?highlight=series%20str%20len#pandas.Series.str.len)